File: src/AuthenticationModules.CS/AuthenticationModuleHelper.cs
using System;
using System.IO;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace AuthenticationModules
{
    static class AuthenticationModuleHelper
    {
        public static void PickUsersFileAndSaveToSetting()
        {
            using (var dialog = new OpenFileDialog())
            {
                dialog.Title = "ユーザーリストファイルを選択してください";
                dialog.FileName = MySettings.Default.UsersFilePath;
                dialog.Filter = "CSVファイル|*.csv";

                if (dialog.ShowDialog() == DialogResult.OK)
                {
                    MySettings.Default.UsersFilePath = dialog.FileName;
                    MySettings.Default.Save();
                }
            }
        }

        public static User[] ReadUsersFile()
        {
            var filePath = MySettings.Default.UsersFilePath;

            if (string.IsNullOrEmpty(filePath))
                return null;

            var text = File.ReadAllText(filePath, Encoding.GetEncoding("shift_jis"));

            return text.Split(new[] { Environment.NewLine }, StringSplitOptions.None).Select(x => x.Split(',')).Where(x => x.Length == 2).Select(x => new User(x[0], x[1])).ToArray();
        }
    }
}

File: src/AuthenticationModules.VB/AuthenticationModuleHelper.vb
Imports System.IO
Imports System.Text

Friend Class AuthenticationModuleHelper

    Public Shared Sub PickUsersFileAndSaveToSetting()

        Using Dialog As New OpenFileDialog

            Dialog.Title = "ユーザーリストファイルを選択してください"
            Dialog.FileName = MySettings.Default.UsersFilePath
            Dialog.Filter = "CSVファイル|*.csv"

            If Dialog.ShowDialog() = DialogResult.OK Then

                MySettings.Default.UsersFilePath = Dialog.FileName
                MySettings.Default.Save()
            End If
        End Using
    End Sub

    Public Shared Function ReadUsersFile() As User()

        Dim FilePath = MySettings.Default.UsersFilePath

        If String.IsNullOrEmpty(FilePath) Then Return Nothing

        Dim Text As String = File.ReadAllText(FilePath, Encoding.GetEncoding("shift_jis"))

        Return Split(Text, Environment.NewLine) _
            .Select(Function(x) Split(x, ",")) _
            .Where(Function(x) x.Length = 2) _
            .Select(Function(x) New User(x(0), x(1))) _
            .ToArray()
    End Function
End Class
